main.tf

```diff
@@ -140,7 +150,67 @@ data "aws_iam_policy_document" "s3_access_for_sftp_users" {
     ]
 
     resources = [
-      join("", data.aws_s3_bucket.landing[*].arn)
+      "arn:aws:s3:::${var.s3_bucket_name}",
+      "arn:aws:s3:::${var.s3_bucket_name}/${each.value}/*"
```

Comment:
Instead of putting in `aws` here, let's use the data source so we can allow govcloud to use this module too

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition